### PR TITLE
Create a Red Hat Enterprise Linux 8 End to End test.

### PR DESCRIPTION
**Pull request by darobs**
Run the E2E tests for Red Hat Enterprise Linux 8.

This runs the same set as all recent OSs - this runs all 29 E2E tests on a RHEL8 server.

## Azure IoT Edge PR checklist:

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines and Best Practices
- [x] I have read the [contribution guidelines](https://github.com/azure/iotedge#contributing).
- [x] Title of the pull request is clear and informative.
- [x] Description of the pull request includes a concise summary of the enhancement or bug fix.

### Testing Guidelines
- [x] Pull request includes test coverage for the included changes.
- Description of the pull request includes 
	- [x] concise summary of tests added/modified
	- [ ] local testing done.

### CODE REVIEW

**Review by gauravIoTEdge — DISMISSED** [review]

File: builds/e2e/e2e.yaml [hunk 1]
```diff
@@ -254,6 +254,28 @@ jobs:
...
 
+  ################################################################################
+  - job: redhat8_amd64
+  ################################################################################
+    displayName: RedHat8 amd64
+
+    pool:
+      name: $(pool.linux.name)
+      demands:
+      - ImageOverride -equals agent-aziotedge-rhel-8-msmoby
```
Comment (yophilav) [inline builds/e2e/e2e.yaml:265]: I thought we have an issue with the disk size for the build. Is it no longer the case?

File: test/Microsoft.Azure.Devices.Edge.Test.Common/linux/PackageManagement.cs [hunk 2]
```diff
@@ -79,15 +93,30 @@ public string[] GetInstallCommandsFromMicrosoftProd(Option<Uri> proxy)
...
-                    $"yum updateinfo",
-                    $"yum install --yes aziot-edge",
-                    "pathToSystemdConfig=$(systemctl cat aziot-edge | head -n 1)",
-                    "sed 's/=on-failure/=no/g' ${pathToSystemdConfig#?} > ~/override.conf",
-                    "sudo mv -f ~/override.conf ${pathToSystemdConfig#?}",
-                    "sudo systemctl daemon-reload"
+                SupportedPackageExtension.Rpm => this.os switch {
+                    "centos" => new[]
+                    {
+                        $"rpm -iv --replacepkgs https://packages.microsoft.com/config/{this.os}/{this.version}/packages-microsoft-prod.rpm",
```
Comment (yophilav) [inline test/Microsoft.Azure.Devices.Edge.Test.Common/linux/PackageManagement.cs:99]: Glad to see this works for AMD64.

File: test/Microsoft.Azure.Devices.Edge.Test.Common/linux/PackageManagement.cs [hunk 3]
```diff
@@ -105,10 +134,11 @@ public string[] GetInstallCommandsFromMicrosoftProd(Option<Uri> proxy)
             },
             SupportedPackageExtension.Rpm => new[]
             {
-                "yum remove -y --remove-leaves aziot-edge",
-                "yum remove -y --remove-leaves aziot-identity-service",
-                "yum remove -y --remove-leaves iotedge",
-                "yum remove -y --remove-leaves libiothsm-std",
+                "yum remove -y aziot-edge",
+                "yum remove -y aziot-identity-service",
```
Comment (yophilav) [inline test/Microsoft.Azure.Devices.Edge.Test.Common/linux/PackageManagement.cs:138]: What about `aziot-certd`, `aziot-identityd`, etc? Are those automatically removed with the uninstallation of the `aziot-edge`?

File: builds/e2e/e2e.yaml (hunk 1 above)
Comment (darobs) [reply builds/e2e/e2e.yaml:265]: Yes, the default host size for the home directory was only 1G, which was not enough space to do a `dotnet restore` during the build.  This is fixed on the build host by running `lvresize -r -L+5G rootvg/homelv`.

File: test/Microsoft.Azure.Devices.Edge.Test.Common/linux/PackageManagement.cs (hunk 3 above)
Comment (darobs) [reply test/Microsoft.Azure.Devices.Edge.Test.Common/linux/PackageManagement.cs:138]: The `--remove-leaves` option was taking take of the dependent packages before this changes, but in Red Hat 8, "yum" is an alias to "dnf" and this option didn't work.  I should achieve the same by invoking `yum autoremove -y`.

**Review by damonbarry — DISMISSED** [review]

File: test/Microsoft.Azure.Devices.Edge.Test.Common/linux/PackageManagement.cs [hunk 4]
```diff
@@ -47,14 +47,28 @@ public string[] GetInstallCommandsFromLocal(string path)
...
-                    "sed 's/=on-failure/=no/g' ${pathToSystemdConfig#?} > ~/override.conf",
-                    "sudo mv -f ~/override.conf ${pathToSystemdConfig#?}",
-                    "sudo systemctl daemon-reload"
+                SupportedPackageExtension.Rpm => this.os switch {
+                    "centos" => new[]
+                    {
+                        "set -e",
+                        $"rpm --nodeps -i {string.Join(' ', packages)}",
+                        "pathToSystemdConfig=$(systemctl cat aziot-edged | head -n 1)",
+                        "sed 's/=on-failure/=no/g' ${pathToSystemdConfig#?} > ~/override.conf",
```
[the remark this answers is not in the document]
Comment (darobs) [reply test/Microsoft.Azure.Devices.Edge.Test.Common/linux/PackageManagement.cs:56]: We overwrite ~/override.conf every time we invoke this. Also, I realized I named the conf file `overrides.conf`, which works, but is not the default naming convention. It should not interfere with a user's default `override.conf` created when you run `systemctl edit`.

[the remark this answers is not in the document]
Comment (darobs) [reply test/Microsoft.Azure.Devices.Edge.Test.Common/linux/PackageManagement.cs:56]: Updated PR to remove `overrides.conf` on package uninstall.

**Review by yophilav — APPROVED** [review]

**Pull request merged**